From 66fc68b559e30019f25874142e39057c942ca4d5 Mon Sep 17 00:00:00 2001
From: Marian Chereji <marian.chereji@nxp.com>
Date: Wed, 15 Jun 2016 12:39:12 +0300
Subject: [PATCH 535/666] dpa_offload/dpa_stats: Fix IOCTL return code in case
 of error

The dpa_stats wrapper was not signaling an error when an IOCTL command
was not recognized. It will now return a EINVAL error code.

Signed-off-by: Marian Chereji <marian.chereji@nxp.com>
[Original patch taken from SDK-V2.0-1703]
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/staging/fsl_dpa_offload/wrp_dpa_stats.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/staging/fsl_dpa_offload/wrp_dpa_stats.c b/drivers/staging/fsl_dpa_offload/wrp_dpa_stats.c
index b7e6cc0..151d6c4 100644
--- a/drivers/staging/fsl_dpa_offload/wrp_dpa_stats.c
+++ b/drivers/staging/fsl_dpa_offload/wrp_dpa_stats.c
@@ -1739,6 +1739,7 @@ static long wrp_dpa_stats_do_ioctl(struct file *filp,
 	default:
 		log_err("Unsupported ioctl 0x%08x, type 0x%02x, nr 0x%02x\n",
 			cmd, _IOC_TYPE(cmd), _IOC_NR(cmd));
+		ret = -EINVAL;
 		break;
 	}
 	return ret;
-- 
2.7.4

